src/lints/trait_removed_associated_type.ron

```diff
@@ -0,0 +1,54 @@
+SemverQuery(
+    id: "trait_removed_associated_type",
+    human_readable_name: "trait's associated type was removed",
+    description: "A trait's associated type was removed or renamed.",
+    required_update: Major,
+    reference_link: Some("https://doc.rust-lang.org/cargo/reference/semver.html#item-remove"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Trait {
+                        trait_name: name @output
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+
+                        importable_path {
+                            path @output @tag
+                        }
+
+                        associated_type {
+                            associated_type: name @output @tag
+
+                            span_: span {
+                                filename @output
+                                begin_line @output
+                            }
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on Trait {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+
+                        importable_path {
+                            path @filter(op: "=", value: ["%path"])
+                        }
+
+                        associated_type @fold @transform(op: "count") @filter(op: "=", value: ["$zero"]) {
+                            name @filter(op: "=", value: ["%associated_type"])
+                        }
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+        "zero": 0,
+    },
+    error_message: "A public trait's associated type was removed or renamed.",
+    per_result_error_template: Some("trait associated type {{trait_name}}::{{associated_type}}, previously at {{span_filename}}:{{span_begin_line}}"),
```

Comment:
```suggestion
    per_result_error_template: Some("associated type {{trait_name}}::{{associated_type}}, previously at {{span_filename}}:{{span_begin_line}}"),
```
it could go either way, let's err on the side of a bit more brief